Chat App/View/Pub/PubRegisterViewController.swift

```diff
@@ -0,0 +1,384 @@
+        view.addSubview(registerButton)
+        
+        registerButton.snp.makeConstraints { make in
+            make.top.equalTo(confirmPasswordMessageLabel.snp.bottom).offset(60)
+            make.centerX.equalToSuperview()
+            make.left.right.equalToSuperview().inset(55)
+            make.height.equalTo(40)
+        }
+    }
+    
+    func createLoginButton() {
+        loginButton.translatesAutoresizingMaskIntoConstraints = false
+        loginButton.setTitle(Constants.PubStrings.Generic.loginButtonTitle, for: .normal)
+        loginButton.backgroundColor = .black
+        loginButton.layer.cornerRadius = 9.0
+        loginButton.addTarget(self, action: #selector(self.loginButtonTapped), for: .touchUpInside)
+        view.addSubview(loginButton)
+        
+        loginButton.snp.makeConstraints { make in
+            make.top.equalTo(registerButton.snp.bottom).offset(15)
+            make.centerX.equalToSuperview()
+            make.left.right.equalToSuperview().inset(55)
+            make.height.equalTo(40)
+        }
+    }
+    
+    func createLoadingIndicator() {
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(activityIndicator)
+        
+        activityIndicator.snp.makeConstraints { make in
+            make.centerX.centerY.equalToSuperview()
+        }
+    }
+}
+
+extension PubRegisterViewController {
+    
+    func setupTextChangedHandlers() {
+        let usernameValid = usernameField
+            .rx
+            .text
+            .observe(on: MainScheduler.asyncInstance)
+            .distinctUntilChanged()
+            .throttle(.milliseconds(inputThrottleInMilliseconds), scheduler: MainScheduler.instance)
+            .map { [unowned self] in
+                self.viewModel.inputs.verifyUserInput(userInput: $0!)
+            }
+        
+        usernameValid
+            .skip(1)
+            .subscribe(onNext: { [weak self] in
+                guard let self = `self` else { return }
+                if $0 {
+                    self.viewModel.inputs.verifyUsernameAvailability(userInput: self.usernameField.text!)
+                } else {
+                    self.usernameMessageLabel.textColor = .red
+                    self.usernameMessageLabel.text = Constants.PubStrings.Warnings.usernameInvalidMessage
+                }
+            })
+            .disposed(by: disposeBag)
+
+        let passwordValid = passwordField
+            .rx
+            .text
+            .observe(on: MainScheduler.asyncInstance)
+            .distinctUntilChanged()
+            .throttle(.milliseconds(inputThrottleInMilliseconds), scheduler: MainScheduler.instance)
+            .map { [unowned self] in
+                self.viewModel.inputs.verifyUserInput(userInput: $0!)
+            }
+        
+        passwordValid
+            .skip(1)
+            .subscribe(onNext: { [weak self] in
+                guard let self = `self` else { return }
+                if $0 {
```

Comment:
instead of using $0 for multiline block, create a variable before `in` so `isValid in`

Reply:
This issue has been addressed in the most recent commits.